sdk/textanalytics/ai-text-analytics/test/apiKey.spec.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { assert, expect, use as chaiUse } from "chai";
+import chaiPromises from "chai-as-promised";
+chaiUse(chaiPromises);
+
+import { Recorder, env } from "@azure/test-utils-recorder";
+
+import { createRecordedClient } from "./utils/recordedClient";
+import { TextAnalyticsClient, TextAnalyticsApiKeyCredential } from "../src/index";
+import { isSuccess } from "./utils/resultHelper";
+import { WebResource } from "@azure/core-http";
+
+const testDataEn = [
+  "I had a wonderful trip to Seattle last week and even visited the Space Needle 2 times!",
+  "Unfortunately, it rained during my entire trip to Seattle. I didn't even get to visit the Space Needle",
+  "I went to see a movie on Saturday and it was perfectly average, nothing more or less than I expected.",
+  "I didn't like the last book I read at all."
+];
+
+describe("TextAnalyticsApiKeyCredential", () => {
+  it("credential signRequest throws on undefined webResource", async () => {
+    const credential = new TextAnalyticsApiKeyCredential(env.SUBSCRIPTION_KEY);
+
+    return assert.isRejected(
+      credential.signRequest((undefined as unknown) as WebResource),
```

Comment:
this is a pretty goofy chain, can we do

```suggestion
      credential.signRequest(undefined as any),
```

It's more explicitly breaking the type system this way.